src/backend/joanie/api.py

```diff
@@ -0,0 +1,80 @@
+"""
+API endpoints
+"""
+from django.shortcuts import get_object_or_404
+
+from rest_framework import generics, pagination, permissions, views
+from rest_framework.response import Response
+
+from joanie.core import exceptions, models
+
+from . import serializers
+
+
+class CourseProductsAvailableListView(views.APIView):
+    """
+    For a course, return all products available with its course runs
+    """
+
+    def get(self, request, code):
+        """Return a list of all products available for a course"""
+        course = get_object_or_404(models.Course, code=code)
+        return Response(serializers.ProductSerializer(course.products, many=True).data)
+
+
+class OrderPagination(pagination.PageNumberPagination):
+    """Order pagination to display no more than 100 orders per page"""
+
+    ordering = "-created_on"
+    page_size = 100
+
+
+class OrdersView(generics.ListAPIView):
+    """
+    API view allows to get all orders or create a new one with lms enrollments for a user.
+
+    GET /api/orders/
+        Return list of all orders for a user with pagination
+
+    POST /api/orders/ with expected data:
+        - id: product uid
+        - resource_links: list of resource_links of all course runs selected
+        Return new order just created
+    """
+
+    serializer_class = serializers.OrderSerializer
+    pagination_class = OrderPagination
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        user = models.User.objects.get_or_create(username=self.request.user.username)[0]
```

Comment:
I wonder what happens if the user is anonymous? Make sure to battle test your API with anonymous requests in tests.

Reply:
I already test views without authentication:
-> https://github.com/openfun/joanie/pull/7/files#diff-fbac6954674d1b9b1b028c06f576fc0d8dab3521cbec3c8a7c3c3cc75b4e0919R221
-> https://github.com/openfun/joanie/pull/7/files#diff-fbac6954674d1b9b1b028c06f576fc0d8dab3521cbec3c8a7c3c3cc75b4e0919R645

DRF deals with it thanks to:
https://github.com/openfun/joanie/pull/7/files#diff-54898b9fe6e3561ab95c961105af21d6298de802e306a9ec6339020590e4f000R48